### PR TITLE
Fixed issue 19255: PHP8.1 + debug : survey with only stardate broke survey listing

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: check if is null when needed
Dev: move != to !empty

### CODE REVIEW

File: application/models/Survey.php [hunk 1]
```diff
@@ -1154,23 +1154,23 @@ public function getRunning()
...
         if ($this->active === 'N') {
             $running = '<a href="' . App()->createUrl('/surveyAdministration/view/surveyid/' . $this->sid) . '" class="survey-state" data-bs-toggle="tooltip" title="' . gT('Inactive') . '"><i class="ri-stop-fill text-secondary me-1"></i>' . gT('Inactive') . '</a>';
-        } elseif ($this->expires != '' || $this->startdate != '') {
+        } elseif (!empty($this->expires) || !empty($this->startdate)) {
             // If it's active, then we check if not expired
             // Time adjust
             $sNow    = date("Y-m-d H:i:s", strtotime((string) Yii::app()->getConfig('timeadjust'), strtotime(date("Y-m-d H:i:s"))));
-            $sStop   = ($this->expires != '') ? date("Y-m-d H:i:s", strtotime((string) Yii::app()->getConfig('timeadjust'), strtotime($this->expires))) : null;
-            $sStart  = ($this->startdate != '') ? date("Y-m-d H:i:s", strtotime((string) Yii::app()->getConfig('timeadjust'), strtotime($this->startdate))) : null;
+            $sStop   = !empty($this->expires) ? date("Y-m-d H:i:s", strtotime((string) Yii::app()->getConfig('timeadjust'), strtotime($this->expires))) : null;
```
Comment (Shnoulle) [inline application/models/Survey.php:1161]: Maybe it's more simple to do whole action one time only.

**Comment by mfavetti** [comment]
Looks good to me, even more thorough of a fix than I was looking at.

**Comment by Shnoulle** [comment]
OK, i remove code duplication (4 times) and create a static private function
More clear now ?

**Comment by Shnoulle** [comment]
@mfavetti i can not assign you as reviewer, but happy to have your review :)

**Review by gabrieljenik — APPROVED** [review]
Code structure looks good.

Some php unit for `shiftedDateTime($datetime)` would be good.
Let me know if you handle that or prefer me to do it.
Thanks

**Comment by Shnoulle** [comment]
> Code structure looks good.
> 
> Some php unit for `shiftedDateTime($datetime)` would be good. Let me know if you handle that or prefer me to do it. Thanks

It's more a shortcut to https://github.com/LimeSurvey/LimeSurvey/blob/eb27f86043718396dd71bcbde31b898683114615/application/helpers/common_helper.php#L2569 no ?

I think you ask for a new helper ;)

**Comment by gabrieljenik** [comment]
> > Code structure looks good.
> > Some php unit for `shiftedDateTime($datetime)` would be good. Let me know if you handle that or prefer me to do it. Thanks
> 
> It's more a shortcut to
> 
> https://github.com/LimeSurvey/LimeSurvey/blob/eb27f86043718396dd71bcbde31b898683114615/application/helpers/common_helper.php#L2569
> 
> no ?
> I think you ask for a new helper ;)

OK, no worries

**Comment by Shnoulle** [comment]
> > > Code structure looks good.
> > > Some php unit for `shiftedDateTime($datetime)` would be good. Let me know if you handle that or prefer me to do it. Thanks
> > 
> > 
> > It's more a shortcut to
> > https://github.com/LimeSurvey/LimeSurvey/blob/eb27f86043718396dd71bcbde31b898683114615/application/helpers/common_helper.php#L2569
> > 
> > no ?
> > I think you ask for a new helper ;)
> 
> OK, no worries

You see another Model where we need shiftedDateTime ?

**Comment by gabrieljenik** [comment]
> You see another Model where we need shiftedDateTime ?

Actually, now that you say so I realize this:
1- Would be usefull on tokens as well
2- The new method maybe should be on a helper ? 
     https://github.com/LimeSurvey/LimeSurvey/pull/3629/files#diff-0268c3e4648700cef058f99725fd5364a285bd35701d30cfda885a81ce1c86a4R2386
     This method doesn't use any attribute from its class and it could be reused by some others.

**Comment by gabrieljenik** [comment]
> @gabrieljenik Ok to create a Dev issue about a new Helper ? Merge this one and i take the helper with
> 
> * Usage on Tokens (or elsewhere)
> * Create a test
> 
> (or you take if you prefer).

Yes, it is OK to create it. I think it belongs to another ticket.
You take it if you wish. If not, later tibor will assign it when he considers

**Comment by Shnoulle** [comment]
https://bugs.limesurvey.org/view.php?id=19286

**Pull request merged**